armi/reactor/converters/geometryConverters.py

```diff
@@ -789,6 +780,105 @@ def _createRadialThetaZone(
+        categoryByNuclide = {}        
+        for category, nucs in nuclideCategories.items():
+            for nuc in nucs:
+                categoryByNuclide[nuc] = category
+        
+        # Set up some dictionary contains for the temperatures and
+        # number densities for each of the categories. These are assigned
+        # to each new component below after they are populated.
+        tempByNuclideCategory = collections.defaultdict(float)
+        ndensByNuclideCategory = collections.defaultdict(dict)
+        volumeFracsByNuclideCategory = collections.defaultdict(float)
+        for category in nuclideCategories:
+            ndensByNuclideCategory[category] = collections.defaultdict(float)      
+        
+        print(newBlock)
+        for b in self.blockMap[newBlock]:            
+            # Calculate the average temperatures of each category of nuclides.
+            for category, nucs in nuclideCategories.items():
+                nvt, nv = getBlockNuclideTemperatureAvgTerms(b, nucs)
+                # Update the nvt and nv arrays to remove any indices in nv where the division 
+                # of nvt/nv would result in a `nan`.
+                nnvt = []
+                nnv = []
+                for i, nuc in enumerate(nucs):
+                    if nv[i] == 0.0:
+                        continue
+                    nnvt.append(nvt[i])
+                    nnv.append(nv[i])
+                
+                tempByNuclideCategory[category] = numpy.mean(numpy.array(nnvt)/numpy.array(nnv))
+                
+            # Determine the volume fractions of the each component/category and the number densities
+            for c in b:
+                
+                # This is the volume fraction of the component within the new homogenized block,
+                # taking into account the fact that the block that this component belongs to
+                # may have been sliced axially based on the mesh boundaries.
+                volFracInNewBlock = c.getVolumeFraction() * self.blockVolFracs[newBlock][b]
+                
+                ndens = c.getNumberDensities()
+                for nuc, category in categoryByNuclide.items():
+                    nucNumberDensity = ndens.get(nuc, 0.0)
+                    ndensByNuclideCategory[category][nuc] += nucNumberDensity * volFracInNewBlock
+                
+                
+                #### THIS IS WRONG -- I am stumped how to get the right volume fractions for the
+                # new components within the homogenized block so that mass is preserved....
+                if c.getName() == "coolant":
+                    volumeFracsByNuclideCategory["coolant"] += volFracInNewBlock
+                elif "fuel" in c.getName():
+                    volumeFracsByNuclideCategory["fuel"] += volFracInNewBlock 
+                else:
+                    volumeFracsByNuclideCategory["structure"] += volFracInNewBlock 
```

Comment:
@ntouran - I am stumped here. Trying to create components that are based on the categorization of the nuclides so that the (1) temperatures of the nuclides can be preserved from XS generation and (2) so that masses are all conserved when converting between the reactors.

I reused the `getBlockNuclideTemperatureAvgTerms` from the XS group manager above to get the correct temperatures, but I am not sure what to do with calculating the volume fractions.

Basically we have:

newBlock represents the homogenization of bunch of blocks in the core. These other blocks aren't necessarily full blocks because they can get cut off based on the user-defined mesh during the conversion. I believe I am calculating the scaling number densities correctly, but I don't know how to determine which "fraction" of a component belongs to which category of nuclides.

First I tried to scale by number densities rather than using the names of the components, but that didn't work. Some peer programming/insight would be super helpful here.